static/src/javascripts/projects/commercial/modules/hosted/onward.js

```diff
@@ -1,54 +1,61 @@
 // @flow
 import config from 'lib/config';
-import mediator from 'lib/mediator';
 import fetchJson from 'lib/fetch-json';
 import fastdom from 'lib/fastdom-promise';
-import HostedCarousel from 'commercial/modules/hosted/onward-journey-carousel';
+import { initHostedCarousel } from './onward-journey-carousel';
 
-const loadOnwardComponent = (
+const getPlaceholdersFromDom = (dom = document): Array<any> =>
+    [].slice.call(dom.getElementsByClassName('js-onward-placeholder'));
+
+const generateUrlFromConfig = (
+    c: {
+        page: {
+            ajaxUrl: string,
+            pageId: string,
+            contentType: string,
+        },
+    } = config
+): string => `${c.page.ajaxUrl}/${c.page.pageId}/${c.page.contentType.toLowerCase()}/onward.json`;
+
+const insertHTMLfromPlaceholders = (
+    json: { html: string },
+    placeholders: Array<any>
+): void => {
+    placeholders[0].insertAdjacentHTML('beforeend', json.html);
```

Comment:
It is not semantically equivalent with the previous form, but perhaps you _know_ `placeholders` will only ever have one element? If yes then I suggest fixing that in the contract: `placeholder: Element`